src/Akeneo/Pim/Automation/DataQualityInsights/back/Application/ComputeProductsKeyIndicators.php

```diff
@@ -26,17 +26,17 @@ public function __construct(GetLocalesByChannelQueryInterface $getLocalesByChann
     }
 
     /**
-     * @param ProductId[] $productIds
+     * @param ProductIdCollection $productIdCollection
      *
      * @return array
      */
-    public function compute(array $productIds): array
+    public function compute(ProductIdCollection $productIdCollection): array
     {
         $localesByChannel = $this->getLocalesByChannelQuery->getArray();
-        $keyIndicatorsResultsByName = $this->executeAllKeyIndicatorsQueries($productIds);
+        $keyIndicatorsResultsByName = $this->executeAllKeyIndicatorsQueries($productIdCollection);
 
         $productsKeyIndicators = [];
-        foreach ($productIds as $productId) {
+        foreach ($productIdCollection->toArray() as $productId) {
```

Comment:
ProductIdCollection is an Iterator, you can use it directly in a loop
```suggestion
        foreach ($productIdCollection as $productId) {
```

Reply:
Done